docs/upgrade.html

```diff
@@ -19,6 +19,21 @@
 
 <script id="upgrade-template" type="text/x-handlebars-template">
 
+<h4><a id="upgrade_4_1_0" href="#upgrade_4_1_0">Upgrading to 4.1.0 from any version 0.8.x through 4.0.0</a></h4>
+    <h5><a id="upgrade_410_notable" href="#upgrade_410_notable">Notable changes in 4.1.0</a></h5>
+    <ul>
+        <li>A number of deprecated classes, methods, configurations and tools have been removed.
+            <ul>
+                <li><b>Common</b>
+                    <ul>
+                        <li>The <code>kafka.common.errors.NotLeaderForPartitionException</code> class was removed. Please use the <code>kafka.common.errors.NotLeaderOrFollowerException</code> class instead.
```

Comment:
Let's use the full class names `org.apache.kafka.common.errors`.

Also users should not have to _use_ `NotLeaderOrFollowerException`. Instead we can say something like `org.apache.kafka.common.errors.NotLeaderOrFollowerException` is returned if a request could not be processed because the broker is not the leader or follower for a topic partition.

Reply:
> Let's use the full class names `org.apache.kafka.common.errors`.
> 
> Also users should not have to _use_ `NotLeaderOrFollowerException`. Instead we can say something like `org.apache.kafka.common.errors.NotLeaderOrFollowerException` is returned if a request could not be processed because the broker is not the leader or follower for a topic partition.

Sounds good.Thanks for review! Addressed it.